### ISSUE TITLE
PWAs just open a new firefox window

### ISSUE THREAD

**Issue by MaskMan191** [filips123/PWAsForFirefox#395]
### Description

Whenever I open a PWA that I've made, it just opens a new, normal firefox window. 

### Steps to Reproduce

1. open a PWA

### Affected Websites

* ...
* ...

### Environment

* Operating system: Windows 10 Professional
* System architecture: x64?
* Desktop environment: N/A
* Installation method: MSI
* PWAsForFirefox extension version: current
* PWAsForFirefox native version: current
* PWAsForFirefox runtime version: current
* Firefox version: current

**Comment by Dolyfin** [comment]
I have the same issue. Used to work with youtube music fine. Now its just a new firefox window.

**Comment by filips123** [comment]
This is probably duplicate of #388, You can run "Update web apps" from the extension settings to apply the fix.

**Comment by MaskMan191** [comment]
> This is probably duplicate of #388, You can run "Update web apps" from the extension settings to apply the fix.

This is not a duplicate of #388 as I updated to FireFox 117 well before my PWAs stopped working and every update and fix I've tried (including that one) does not work.

**Comment by MaskMan191** [comment]
Did not mean to close, sorry

**Comment by filips123** [comment]
Does it work if you install a new web app into a new profile?

**Comment by ggplayer0** [comment]
I have the exact same problem: A regular, vanilla-looking Firefox window opens with a new tab every time I try to open a web app. After some testing, this bug seems to occur after choosing "Reinstall runtime" in the extension settings. (Maybe this has to do with originally using the .msi file to install PWAsForFirefox? It seems like the profiles are corrupted, idk.)

**A fix/workaround that works for me is to create a new profile, and use that for web apps instead of the default one.** 

After "removing" the Default profile (which apparently only clears its web apps and date), web apps installed through the Default profile now open a window with PWAsForFirefox's custom layout... but the entire page is blank, labelled "New Tab", and I cannot access any pages, including the settings. Neither reinstalling the extension, nor reinstalling manually with the .msi file, nor choosing "Update Web Apps" fixes this problem.

I've attached a runtime log of the Default profile before I "removed" it. I have not been able to open developer tools in it after clearing it.
[console-export-2023-9-12_19-42-15.txt](https://github.com/filips123/PWAsForFirefox/files/12592888/console-export-2023-9-12_19-42-15.txt)

**Comment by DasBabyPixel** [comment]
Same problem here, creating new profiles does not help in my case

**Comment by filips123** [comment]
Please try to launch your web app from console with:

```
firefoxpwa site launch SITE-ID -- --jsconsole
```

You can get the site ID with:

```
firefoxpwa profile list
```

This should display runtime logs, regardless if startup is broken.

---

You can try to remove `chrome` directory from each of web app profiles, and files `_autoconfig.cfg` and `defaults/pref/autoconfig.js` from the runtime directory. You can see where these directories are [here](https://pwasforfirefox.filips.si/resources/installation-directories/#user-data).

**Comment by DasBabyPixel** [comment]
Problem still there
![image](https://github.com/filips123/PWAsForFirefox/assets/43410952/ed4d0250-c7a6-4132-bc9c-3b59e37ca8dd)
![image](https://github.com/filips123/PWAsForFirefox/assets/43410952/b160e703-9a5d-4f86-9169-d10ce3ca30db)

**Comment by MaskMan191** [comment]
> Please try to launch your web app from console with:
> 
> ```
> firefoxpwa site launch SITE-ID -- --jsconsole
> ```
> 
> You can get the site ID with:
> 
> ```
> firefoxpwa profile list
> ```
> 
> This should display runtime logs, regardless if startup is broken.
> 
> You can try to remove `chrome` directory from each of web app profiles, and files `_autoconfig.cfg` and `defaults/pref/autoconfig.js` from the runtime directory. You can see where these directories are [here](https://pwasforfirefox.filips.si/resources/installation-directories/#user-data).

Still there for me as well
![image](https://github.com/filips123/PWAsForFirefox/assets/144282653/2423f66d-4338-4e15-8e18-6ba8d020f34e)

**Comment by filips123** [comment]
Can you send the content of `_autoconfig.cfg` and `defaults/pref/autoconfig.js` inside your runtime directory, and `chrome/pwa/boot.jsm` and `chrome/pwa/chrome.jsm` inside your web app profile directory?

**Comment by MommyHeather** [comment]
Can confirm I had this after updating, choosing *update web apps* did *not* fix it, using a new profile did. To avoid losing my data though, I managed to fix it with *always patch runtime and profile* enabled - this fixed it and all apps remained fixed even after disabling this option.

**Comment by filips123** [comment]
Yeah, this makes sense. I thought "Update all web apps" will also re-patch the runtime, but it doesn't. So, if anyone else has this problem, try enabling "Always patch runtime and profile" and launch web apps. You can later disable it.

**Comment by MommyHeather** [comment]
Yeah, that makes sense - so question is, what does *always patch runtime and profile* do that *update all web apps* doesn't?

**Comment by filips123** [comment]
It always calls [`runtime.patch`](https://github.com/filips123/PWAsForFirefox/blob/main/native/src/components/runtime.rs#L272) (which copied some files to runtime directory) and [`profile.patch`](https://github.com/filips123/PWAsForFirefox/blob/main/native/src/components/profile.rs#L54) (which copies some files to profile directory) when launching a web app. Normally, the native program should detect when these files have changed (when a new version is installed) and only copy them if needed, but for some reason, it doesn't always detect it for some users.

**Comment by MommyHeather** [comment]
> It always calls [`runtime.patch`](https://github.com/filips123/PWAsForFirefox/blob/main/native/src/components/runtime.rs#L272) (which copied some files to runtime directory) and [`profile.patch`](https://github.com/filips123/PWAsForFirefox/blob/main/native/src/components/profile.rs#L54) (which copies some files to profile directory) when launching a web app. Normally, the native program should detect when these files have changed (when a new version is installed) and only copy them if needed, but for some reason, it doesn't always detect it for some users.

Perhaps the *update all web apps* feature should also call these then?

**Comment by filips123** [comment]
Maybe, but it's not really updating web apps. A separate "Patch all profiles (and runtime)" button might be better.

**Comment by filips123** [comment]
I'll close this issue. If you still have problems after trying [this solution](https://github.com/filips123/PWAsForFirefox/issues/395#issuecomment-1721514381), please let me know. I'll probably also add "Patch all profiles (and runtime)" button in the next release.

**Comment by MaskMan191** [comment]
> I'll close this issue. If you still have problems after trying [this solution](https://github.com/filips123/PWAsForFirefox/issues/395#issuecomment-1721514381), please let me know. I'll probably also add "Patch all profiles (and runtime)" button in the next release.

Yeah, it's still not working for me.

**Comment by filips123** [comment]
Please send the content of `_autoconfig.cfg` and `defaults/pref/autoconfig.js` inside your runtime directory, and `chrome/pwa/boot.jsm` and `chrome/pwa/chrome.jsm` inside your web app profile directory.

Also, can you go to the extension settings and to about this project, and copy all version information?

**Comment by AxelTerizaki** [comment]
Hello.

I have a similar issue, except it just opens the PWA twice (or more) no matter how many times I click on the icon.

I have several PWAs and they're all fine. When I click on their icon they switch to their window if they're already open.

However, with a PWA I made for deck.blue, everytime I click on the app in my dock, it doesn't switch to it, but launches a new window. A new PWA window though, not a new firefox window. 

What can I provide to help diagnose the issue?

**Comment by filips123** [comment]
So, this happens only with one specific PWA, while others are fine? Well, the first thing you can do to diagnose this is to check if "When launching a web app that is already opened" is set to "Focus the existing window" in the settings. You may also need to restart PWAs to apply the setting. If that is set correctly but doesn't work, I'll check if there are any other things you can diagnose.

**Comment by AxelTerizaki** [comment]
Well, nevermind.

There must have been an update somehow or maybe I restarted my system and it fixed it, but the issue isn't there anymore.

Sorry for the trouble :)
